File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/course/SectionItemAdminService.java
package ma.zs.zyn.service.facade.admin.course;

import java.util.List;
import ma.zs.zyn.bean.core.course.SectionItem;
import ma.zs.zyn.dao.criteria.core.course.SectionItemCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface SectionItemAdminService extends  IService<SectionItem,SectionItemCriteria>  {

    List<SectionItem> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}

File: frontend/src/app/shared/service/admin/course/SectionItemAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {SectionItemDto} from 'src/app/shared/model/course/SectionItem.model';
import {SectionItemCriteria} from 'src/app/shared/criteria/course/SectionItemCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class SectionItemAdminService extends AbstractService<SectionItemDto, SectionItemCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): SectionItemDto {
        return new SectionItemDto();
    }

    public constrcutCriteria(): SectionItemCriteria {
        return new SectionItemCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/sectionItem/';
    }
}
